ssl/ssl_local.h

```diff
@@ -1889,6 +1888,16 @@ pitem *pqueue_iterator(pqueue *pq);
 pitem *pqueue_next(piterator *iter);
 size_t pqueue_size(pqueue *pq);
 
+/* Forward-declare struct timeval. On Windows, it is defined in winsock2.h and
+ * Windows headers define too many macros to be included in public headers.
+ * However, only a forward declaration is needed. */
+struct timeval;
+
+struct OPENSSL_timeval {
```

Comment:
I wonder why we're duplicating a system structure.  Would it suffice leave just the forward declaration of `struct timeval` and insist that callers who want to get inside include `winsock2.h`?

I'm concerned about confusion as to which structure should be used where.

Reply:
I replace one`struct OPENSSL_timeval` member by this one, 20 line below
to be able to cut the #include dependency to `winsock.h`

Else, I will have to add it to every dtls source file in this folder.

See windows definition of [`struct timeval`](https://docs.microsoft.com/en-us/windows/win32/api/winsock/ns-winsock-timeval)

My quick reading of #4188 make me understand that you hit this issue.

Reply:
Oups, with this change, break the call to BIO_CTRL_DGRAM_SET_NEXT_TIMEOUT,
who is expecting a pointer to a 'struct timeval' object.

Reply:
I try to fix this in the simplest way.
There no need to store a true `timeval` struct as member of `DTLS1_STATE`